src/App.tsx

```diff
@@ -8,7 +8,25 @@
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(true);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [allTodos, setAllTodos] = useState<Todo[]>();
+
+  useEffect(() => {
+    if (!allTodos) {
+      getTodos().then(setAllTodos, setTodos);
```

Comment:
The `getTodos` function is being used incorrectly. The `then` method should be used to set `allTodos` and handle any errors separately. Consider using `.then(setAllTodos).catch(errorHandler)` to handle errors properly.